OHHTTPStubs/UnitTests/Test Suites/NSURLSessionTests.m

```diff
@@ -241,13 +255,47 @@ - (void)test_NSURLSessionDefaultConfig_notFollowingRedirects
         NSURLSession *session = [NSURLSession sessionWithConfiguration:config delegate:self delegateQueue:nil];
 
         NSDictionary* json = @{@"Success": @"Yes"};
-        [self _test_redirect_NSURLSession:session jsonForStub:json completion:^(NSError *errorResponse, NSHTTPURLResponse *redirectResponse, id jsonResponse) {
+
+        NSDictionary* jsonBody = @{@"Testing":@"OneTwoThree"};
+        NSError *error;
+        NSData * dataToPost = [NSJSONSerialization dataWithJSONObject:jsonBody options:NSJSONWritingPrettyPrinted error:&error];
+        XCTAssertNil(error, @"Unexpected error recieved");
+
+        [self _test_redirect_NSURLSession:session jsonForStub:json httpMethod:@"GET" httpBody:nil httpStatus:301 completion:^(NSError *errorResponse, NSHTTPURLResponse *redirectResponse, id jsonResponse) {
             XCTAssertNil(errorResponse, @"Unexpected error");
             XCTAssertNotNil(redirectResponse, @"Redirect response should have been received");
             XCTAssertEqual(301, [redirectResponse statusCode], @"Expected 301 redirect");
             XCTAssertNil(jsonResponse, @"Unexpected data received");
         }];
 
+        [self _test_redirect_NSURLSession:session jsonForStub:json httpMethod:@"POST" httpBody:dataToPost httpStatus:302 completion:^(NSError *errorResponse, NSHTTPURLResponse *redirectResponse, id jsonResponse) {
+            XCTAssertNil(errorResponse, @"Unexpected error");
+            XCTAssertNotNil(redirectResponse, @"Unexpected redirect response received");
+            XCTAssertEqual(302, [redirectResponse statusCode], @"Expected 302 redirect");
+            XCTAssertNil(jsonResponse, @"Unexpected data received");
+        }];
```

Comment:
So for example here if we tweak our `test_redirect_NSURLSession…` method so that it returns a response body derived from the `dataToPost`, we can now check if the `jsonResponse` — which should be derived from that `g(f(dataToPost))` that we initially sent — matches what is expected.

If our redirect lost the `httpBody` during that redirect, that means that we would lose the response derived from it and the assertion would fail.

---

That's just a random idea, I'm on the train rn with little data so haven't had the leisure of looking into it in details, I hope you'll see the idea.